cxx/pclean/schema_helper.cc

```diff
@@ -67,6 +74,26 @@ PCleanVariable PCleanSchemaHelper::get_scalarvar_from_path(
   assert(false);
 }
 
+// Returns original_domains, but with the elements corresponding to
+// annotated_ds elements that start with prefix moved to the front.
+std::vector<std::string> reorder_domains(
+    const std::vector<std::string>& original_domains,
+    const std::vector<std::string>& annotated_ds,
+    const std::string& prefix) {
+  std::vector<std::string> output_domains;
+  for (size_t i = 0; i < original_domains.size(); ++i) {
+    if (annotated_ds[i].starts_with(prefix)) {
+      output_domains.push_back(original_domains[i]);
+    }
+  }
+  for (size_t i = 0; i < original_domains.size(); ++i) {
```

Comment:
It looks like this block of code is repeated verbatim, is that intended? I'm finding this function hard to reason about. Could you add a unit test for it?

Reply:
Oh I missed `!annotated_ds[i]` vs. `annotated_ds[i]`, nvm about it being repeated.

Reply:
Added unit test.